src/views/List/List.jsx

```diff
@@ -35,6 +43,21 @@ export default function List({ token }) {
     });
   });
 
+  //added searchInput state value as a dependency so the search criteria updates as soon as the user types
+  useEffect(() => {
+    //the original data set is filtered so we always search from all the list items
+    let searchResults = data.filter((listItem) => {
+      return listItem.name.toLowerCase().includes(searchInput.toLowerCase());
+    });
+    //if search input is empty (the user hasnt typed anything or they removed their input), set the data back to the original list
+    //value === '' ? setCopyOfData(data) : setCopyOfData(searchResults);
+    searchInput === '' ? setCopyOfData(data) : setCopyOfData(searchResults);
+    //if the search results return no entries, set the search Error to display to the user indicating no entries were found
+    searchResults.length < 1
+      ? setSearchError('No List Items Match Your Search')
```

Comment:
great naming conventions here and notes in the code, super readable!
I love that you included the searchError with a very descriptive message to users, great UX choice here 👏